### DISCUSSION TITLE
pgbouncer doesn't process traffic on VIP

### DISCUSSION THREAD

**Discussion by Gruic** [autobase-tech/autobase#288] [Q&A]
Vitas, thanks for the awesome playbook you built!  This thing is awesome.  However I've been struggling quite a bit getting what it built to be consumed by the installer for AAP (Ansible Automation Platform).  I'm running into an unusual issue where if I point to the primary node ip address of a 3 node cluster, I can connect to pgbouncer just fine on 6432 and database operations are all normal.  However if I use the vip-manager's virtual cluster IP address, the port is listening on 6432, however no data is processed and the operation times out.  Additionally I can recreate the same behavior just using psql when attempting to connect to the database.

In my example here, patroni-ha is my cluster virtual IP address.

root@awx1 RedHat]# psql --username=awx --port=6432 --dbname=awx --host=patroni-ha
Password for user awx: 
psql (13.7, server 15.2)
WARNING: psql major version 13, server major version 15.
         Some psql features might not work.
Type "help" for help.

awx=> \d
FATAL:  query_wait_timeout
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Succeeded.
psql (13.7, server 15.2)
WARNING: psql major version 13, server major version 15.
         Some psql features might not work.
----------------------

So I looked up the error and has something to do with pgbouncer configuration.  Has anyone else ran into this issue?  When I do the same series of command on the primary node holding the cluster virtual IP, but using the node's IP itself, everything works great.

**Comment by vitabaks** [comment]
> query_wait_timeout

You may have specified too small pool_size which is not enough for your application, especially if you are using pool_mode: "session".
Always try to use pool_mode = transaction (you need to make sure your application supports transaction mode). This mode allows for more efficient reuse of pool connections.

details: https://github.com/vitabaks/postgresql_cluster/issues/89

**Reply by Gruic** [reply]
I have done this and the Ansible Automation Platform playbook installer has gotten much further.  Thank you for your help -- Almost there!

**Reply by Gruic** [reply]
My last error was a licensing error.  I now have a highly available Ansible Automation Platform v2.3 set up.  Thank you again for all your patience and assistance.

**Comment by vitabaks** [comment]
> thanks for the awesome playbook you built! This thing is awesome.

Thanks!
